providers/standard/src/airflow/providers/standard/operators/hitl.py

```diff
@@ -84,6 +84,9 @@ def __init__(
         self.multiple = multiple
 
         self.params: ParamsDict = params if isinstance(params, ParamsDict) else ParamsDict(params or {})
+        if hasattr(self.params, "filter_params_by_source"):
+            # Params that exist only in Dag level does not make sense to appear in HITLOperator
+            self.params = ParamsDict.filter_params_by_source(self.params, source="task")
```

Comment:
It _seems_ we don't need to check the attribute as we had already cast the `self.params` as `ParamsDict`, then we could directly call the `filter_params_by_source` method?

```suggestion
        self.params = ParamsDict.filter_params_by_source(self.params, source="task")
```

Reply:
It's for backward compatibility, but `hasattr(ParamsDict, "filter_params_by_source")` is more explicit.